src/table_functions.cpp

```diff
@@ -22,7 +25,11 @@ auto static try_value_is_int(Value s, const std::string& method_name, int arg_in
     auto tmp = Number(1);
     try {
         if (s.is_string()) {
-            tmp = std::get<Number>(s.to_number());
+            Value v = s.to_number();
+            if (v == Nil()) {
+                throw std::runtime_error("");
+            }
+            tmp = std::get<Number>(v);
         } else if (!s.is_number()) {
             throw std::runtime_error("");
```

Comment:
You should give an error message here. Also the condition here is always true, otherwise we would have returned from line 18.

Reply:
this case is for s is any type besides string and number, which is an error.
also the throw is only there to trigger the catch